File: Java/StripComments.java
//4 kyu
//https://www.codewars.com/kata/51c8e37cee245da6b40000bd
import java.util.Arrays;
import java.util.stream.Collectors;

public class StripComments {

    public static String stripComments(String text, String[] commentSymbols) {
        String res = Arrays.stream(text.split("\n"))
                .map(line -> getStrippedLine(line, commentSymbols))
                .collect(Collectors.joining());

        return res.substring(0, res.lastIndexOf("\n"));
    }

    private static String getStrippedLine(String line, String[] commentSymbols) {
        int firstSymbolIndex = Integer.MAX_VALUE;

        for (String symbol : commentSymbols) {
            int currentSymbolIndex = line.indexOf(symbol);
            if (currentSymbolIndex >= 0)
                firstSymbolIndex = Math.min(firstSymbolIndex, currentSymbolIndex);
        }

        if (firstSymbolIndex == Integer.MAX_VALUE)
            return rightTrim(line) + "\n";
        return rightTrim(line.substring(0, firstSymbolIndex)) + "\n";
    }

    private static String rightTrim(String s) {
        int i = s.length() - 1;
        while (i >= 0 && Character.isWhitespace(s.charAt(i)))
            i--;

        return s.substring(0, i + 1);
    }
}

File: C#/StripComments.cs
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
public class StripCommentsSolution
{
        public static string StripComments(string text, string[] commentSymbols)
        {
            var commentChars = commentSymbols.SelectMany(s => s.Select(c => c)).ToArray();
            var list = new List<string>();
            foreach (var line in text.Split('\n'))
            {
                var index = line.IndexOfAny(commentChars);
                if (index == -1)
                    list.Add(line.TrimEnd());
                else
                    list.Add(line.Substring(0, index).TrimEnd());
            }
            return string.Join("\n", list);
        }
}